packages/api/tests/fieldCrop.test.js

```diff
@@ -0,0 +1,562 @@
+      let fieldCrop = fakeFieldCrop();
+      fieldCrop.estimated_revenue = 1;
+      fieldCrop.area_used = field.area * 0.25;
+      fieldCrop.estimated_production = 1;
+      postFieldCropRequest(fieldCrop, (err, res) => {
+          console.log(fieldCrop,res.error);
+          expect(res.status).toBe(403);
+          done()
+        },
+        newUser.user_id)
+    });
+  });
+
+  describe('Post crop', () => {
+
+    test('should return 400 status if crop is posted w/o crop_common_name', (done) => {
+      let crop = fakeCrop();
+      delete crop.crop_common_name;
+      postCropRequest(crop, (err, res) => {
+        console.log(crop,res.error);
+        expect(res.status).toBe(400);
+        expect(JSON.parse(res.error.text).error.data.crop_common_name[0].keyword).toBe("required");
+        done()
+      })
+    });
+
+    test('should post and get a valid crop', (done) => {
+      let crop = fakeCrop();
+      crop.crop_common_name = `${crop.crop_specie} - ${crop.crop_genus}`;
+      postCropRequest(crop, (err, res) => {
+        console.log(crop,res.error);
+        expect(res.status).toBe(201);
+        getRequest(`/crop/farm/${farm.farm_id}`,(err,res)=>{
+          console.log(crop,res.body);
+          expect(res.status).toBe(200);
+          expect(res.body[1].crop_common_name).toBe(crop.crop_common_name);
+          done();
+        })
+      })
+    });
+
+    test('should return 403 status if crop is posted by unauthorized user', (done) => {
+      let crop = fakeCrop();
+      crop.crop_common_name = `${crop.crop_specie} - ${crop.crop_genus}`;
+      postCropRequest(crop, (err, res) => {
+        console.log(crop,res.error);
+        expect(res.status).toBe(403);
+        expect(res.error.text).toBe("User does not have the following permission(s): add:crops");
+        done()
+      }, newUser.user_id)
+    });
+
+    test('should return 403 status if crop is posted by newWorker', (done) => {
+      let crop = fakeCrop();
+      crop.crop_common_name = `${crop.crop_specie} - ${crop.crop_genus}`;
+      postCropRequest(crop, (err, res) => {
+        console.log(crop,res.error);
+        expect(res.status).toBe(403);
+        expect(res.error.text).toBe("User does not have the following permission(s): add:crops");
+        done()
+      }, newWorker.user_id)
+    });
+
+  });
+
+  describe('crop_common_name + genus + species uniqueness tests', function(){
+    test('should return 400 status if crop is posted w/o variety name', (done) => {
+      let crop = fakeCrop();
+      crop.crop_common_name = `${crop.crop_specie} - ${crop.crop_genus}`;
+      postCropRequest(crop, (err, res) => {
+        console.log(crop,res.error);
+        expect(res.status).toBe(201);
+        postCropRequest(crop, (err, res) => {
+          console.log(crop,res.error);
+          expect(res.status).toBe(400);
+          expect(JSON.parse(res.error.text).error.routine).toBe("_bt_check_unique");
+          done()
+        })
+      })
+    });
```

Comment:
Consider creating one crop directly on the DB then trying to post. I think that would make this test more readable